torch/_library/custom_ops.py

```diff
@@ -187,6 +191,54 @@ def _qualname(self) -> str:
     def __repr__(self) -> str:
         return f"<CustomOpDef({self._qualname})>"
 
+    @contextmanager
+    def disable_kernel(self, device_type: str, disable: bool = True):
+        """
+        Disable or re-enable an already registered kernel for this custom operator.
+
+        If the kernel is already disabled/enabled, this is a no-op.
+
+        Note:
+            If a kernel is first disabled and then registered, it is disabled until enabled again.
+
+        Args:
+            device_type (str): The device type to disable/enable the kernel for.
+            disable (bool): Whether to disable or enable the kernel.
+        """
+        action = "disable" if disable else "enable"
+        originally_disabled = device_type in self._disabled_kernel
+        if device_type not in self._backend_fns:
+            warnings.warn(
```

Comment:
Log instead of warning, since warning will only issue once?

Reply:
changed to use log now.